code/modules/shuttle/computer.dm

```diff
@@ -75,6 +85,149 @@
 	if(port && (shuttleId == initial(shuttleId) || override))
 		shuttleId = port.id
 
+/obj/structure/machinery/computer/shuttle/ert
+	name = "transport shuttle"
+	desc = "A transport shuttle flight computer."
+	icon_state = "syndishuttle"
+	req_access = list()
+	breakable = FALSE
+	var/disabled = FALSE
+
+/obj/structure/machinery/computer/shuttle/ert/is_disabled()
+	return disabled
+
+/obj/structure/machinery/computer/shuttle/ert/disable()
+	disabled = TRUE
+
+/obj/structure/machinery/computer/shuttle/ert/enable()
+	disabled = FALSE
+
+/obj/structure/machinery/computer/shuttle/ert/tgui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		var/obj/docking_port/mobile/shuttle = SSshuttle.getShuttle(shuttleId)
+		ui = new(user, src, "NavigationShuttle", "[shuttle.name] Navigation Computer")
+		ui.open()
+
+
+/obj/structure/machinery/computer/shuttle/ert/ui_status(mob/user, datum/ui_state/state)
+	. = ..()
+	if(disabled)
+		return UI_UPDATE
```

Comment:
shouldn't this be UI_CLOSE

Reply:
So I have it as UI_UPDATE because I don't want users to be able to interact with the screen (probably overkill, theres about 3 guards here) - but I do want the screen to update when the disabled var is set to false. This way they get an IC error message pop up so they get feedback (rather than clicking on the terminal does nothing).